test/asn1_encode_test.c

```diff
@@ -0,0 +1,641 @@
+/*
+ * Copyright 2017 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the OpenSSL license (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <stdio.h>
+#include <string.h>
+
+#include "internal/asn1t.h"
+#include "internal/numbers.h"
+#include "test_main.h"
+#include "testutil.h"
+
+#ifdef __GNUC__
+# pragma GCC diagnostic ignored "-Wunused-function"
+# pragma GCC diagnostic ignored "-Wformat"
+#endif
+#ifdef __clang_
```

Comment:
It's `__clang__`, not `__clang_`. But it works out, because clang defines even `__GNUC__` and does recognize #pragma GCC. Otherwise it would have to be `#ifdef __clang__ ... #elif defined(__GNUC__) ... #endif`